src/main/cpp/output/VisualizerData.h

```diff
@@ -0,0 +1,40 @@
+#ifndef VISUALIZERDATA_H_INCLUDED
+#define VISUALIZERDATA_H_INCLUDED
+
+/**
+ * @file
+ * Data structure for the visualiser.
+ */
+
+#include <memory>
+#include <string>
+#include <vector>
+#include <boost/property_tree/ptree.hpp>
+#include "pop/Population.h"
+
+namespace stride {
+
+/**
+ * Data structure for the visualiser.
+ * Stores the day-by-day infected count per Town.
+ */
+class VisualizerData
+{
+private:
+	/// Vector of days: Each day is a map from Town names to the amount of infected.
+	std::vector<std::map<std::size_t, unsigned int>> days;
+
+public:
+	/// Register a new day from the given population's current state.
+	void AddDay(const Population& pop);
+
+	/// Get a reference to the vector of days.
+	const std::vector<std::map<std::size_t, unsigned int>>& GetDays() const;
+
+	/// Convert all stored data into a boost property tree, so that it can be converted to JSON.
+	const std::shared_ptr<boost::property_tree::ptree> ToPtree() const;
```

Comment:
Can you remove the `const` qualifier on the return value here? It doesn't do or mean anything.